porcupine/menubar.py

```diff
@@ -282,15 +284,22 @@ def setup():
 
     def update_enabledness(*junk: object) -> None:
         tab = get_tab_manager().select()
+
         parent, child = _split_parent(path)
         menu = get_menu(parent)
         index = _find_item(menu, child)
         if index is None:
             raise LookupError(f"menu item {path!r} not found")
+        if not tab:
+            menu.entryconfig(index, state="disabled")
+            return
+
         menu.entryconfig(index, state=("normal" if callback(tab) else "disabled"))
 
     update_enabledness()
+    get_tab_manager().bind("<<TabFiletypeApplied>>", update_enabledness, add=True)
```

Comment:
This makes the core Porcupine aware of filetypes. Before this PR they were entirely a `plugins/filetypes.py` concept.

But I'm not sure if this is avoidable. I'll think about this a bit and respond after a couple days. Ping me if you don't hear back :)

Reply:
Well, we could create some sort of a.. registry for tracking what events should trigger the `update_enabledness` callback. Honestly not a bad idea IMO, let me throw that together.

Reply:
Basically action consumers have a registry of virtual events that should trigger them to refresh their available actions, and action providers can register events to them.

Reply:
@Akuli 8128316

Reply:
Aren't event bindings somewhat expensive? This is `events_to_trigger_on = bindings` vs  `events_to_trigger_on*menu_items = bindings`

Reply:
Tbh, at this point I just want this merged and I don't really care :D